ghost/collections/src/Collection.ts

```diff
@@ -40,15 +47,50 @@ export class Collection {
         }
     }
 
+    public edit(data: Partial<Collection>) {
```

Comment:
I'm not 100% sure what the purpose of this method is, why is this preferable setting properties on the entity individually/Object.assign?

Reply:
There definitely will be extra validations and logic involved when the Collection entity is edited. With direct setting of properties the logic eventually will be spread all over the place. Also, direct assignment on the entity makes it hard to to reason where the changes to the entity come from.

Reply:
What do you mean by spread all over the place, it would all be contained in the entity in the setters? And how does direct assignment make it harder to reason where changes come from?

Reply:
> What do you mean by spread all over the place, it would all be contained in the entity in the setters?

It would, but right now we don't have setters on the Collection, so setting new properties was done through `Object.assign` which is not easy to reason about when looking through the code to understand which exact fields are modified. This is also what I meant by "direct assignment", using "Object.assign" - without setters or "edit" method we just modify the object without central place to validate etc.